File: 백준/Silver/13909. 창문 닫기/창문 닫기.c
#include <stdio.h>
#include <math.h>

int main(void) {
    long long n;
    if (scanf("%lld", &n) != 1) return 0;

    long long ans = (long long)sqrt((double)n);
    while ((ans + 1) * (ans + 1) <= n) ans++;
    while (ans * ans > n) ans--;

    printf("%lld\n", ans);
    return 0;
}

File: 백준/Silver/13909. 창문 닫기/창문 닫기.java
import java.io.*;
import java.util.*;

public class Main {
    public static void main(String[] args) throws Exception {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        long n = Long.parseLong(br.readLine().trim());
        long ans = (long)Math.sqrt(n);
        while ((ans + 1) * (ans + 1) <= n) ans++;
        while (ans * ans > n) ans--;
        System.out.println(ans);
    }
}